In [7]:
def main():
    import math as m

    LatitudeOne = float(
        input("Enter the latitude of the first location: ").strip())
    LongitudeOne = float(
        input("Enter the longitude of the first location: ").strip())
    # Longitude and Latitude inputs and angle of elevation
    LatitudeTwo = float(
        input("Enter the latitude of the second location: ").strip())
    LongitudeTwo = float(
        input("Enter the longitude of the second location: ").strip())
    Angle = float(input("Enter the angle measure in degrees: ").strip())

    LatitudeDifference = m.abs(LatitudeTwo-LatitudeOne)
    LongitudeDiffernce = m.abs(LongitudeTwo-LongitudeOne)
    # Converts longitude to miles
    LongitudeConversion = 69.172 * \
        m.cos(m.radians((LatitudeTwo*.5+LatitudeOne*.5)))

    # Latitude*69 converts latitude to miles    #Calculates the slope distance
    SlopeDistanceMiles = m.sqrt(
        (LatitudeDifference*69)**2+(LongitudeDiffernce*LongitudeConversion)**2)
    # Converts slope distance from miles to feet
    SlopeDistanceFeet = SlopeDistanceMiles*5280
    ElevationChangeFeet = SlopeDistanceFeet * \
        m.tan(m.radians(Angle))  # Calculates elevation change in feet

    print(f"The slope distance is {SlopeDistanceFeet:.1f} feet.")
    # Prints outputs
    print(f"The change in elevation is {ElevationChangeFeet:.1f} feet.")


if __name__ == '__main__':
    main()

AttributeError: module 'math' has no attribute 'abs'

In [12]:
def main_test():
    import math as m

    LatitudeOne = 40.431968
    LongitudeOne = -86.922644
    # Longitude and Latitude inputs and angle of elevation
    LatitudeTwo = 40.432703
    LongitudeTwo = -86.92344
    Angle = 4.93

    LatitudeDifference = abs(LatitudeTwo-LatitudeOne)
    LongitudeDiffernce = abs(LongitudeTwo-LongitudeOne)
    # Converts longitude to miles
    LongitudeConversion = 69.172 * \
        m.cos(m.radians((LatitudeTwo*.5+LatitudeOne*.5)))

    # Latitude*69 converts latitude to miles    #Calculates the slope distance
    SlopeDistanceMiles = m.sqrt(
        (LatitudeDifference*69)**2+(LongitudeDiffernce*LongitudeConversion)**2)
    # Converts slope distance from miles to feet
    SlopeDistanceFeet = SlopeDistanceMiles*5280
    ElevationChangeFeet = SlopeDistanceFeet * \
        m.tan(m.radians(Angle))  # Calculates elevation change in feet

    print(f"The slope distance is {SlopeDistanceFeet:.1f} feet.")
    # Prints outputs
    print(f"The change in elevation is {ElevationChangeFeet:.1f} feet.")


if __name__ == '__main__':
    main_test()
print("")

The slope distance is 347.4 feet.
The change in elevation is 30.0 feet.



Enter the latitude of the first location:  40.431968
Enter the longitude of the first location: -86.922644
Enter the latitude of the second location:  40.432703
Enter the longitude of the second location: -86.92344
Enter the angle measure in degrees: 4.93
The slope distance is 347.4 feet. //348.7
The change in elevation is 30.0 feet. //30.00
